overlord/auth/auth.go

```diff
@@ -81,3 +82,30 @@ func User(st *state.State, id int) (*UserState, error) {
 	}
 	return nil, fmt.Errorf("invalid user")
 }
+
+// Authenticator returns MacaroonAuthenticator for current authenticated user represented by UserState
+func (us *UserState) Authenticator() *MacaroonAuthenticator {
+	return newMacaroonAuthenticator(us.Macaroon, us.Discharges)
+}
+
+// MacaroonAuthenticator is a store authenticator based on macaroons
+type MacaroonAuthenticator struct {
+	Macaroon   string
+	Discharges []string
+}
+
+func newMacaroonAuthenticator(macaroon string, discharges []string) *MacaroonAuthenticator {
+	return &MacaroonAuthenticator{
+		Macaroon:   macaroon,
+		Discharges: discharges,
+	}
+}
+
+// Authenticate will add the store expected Authorization header for macaroons
+func (ma *MacaroonAuthenticator) Authenticate(r *http.Request) {
+	value := fmt.Sprintf(`Macaroon root="%s"`, ma.Macaroon)
+	for _, discharge := range ma.Discharges {
+		value = fmt.Sprintf(`%s, discharge="%s"`, value, discharge)
```

Comment:
Note you don't want the first `%s` with the logic above. The prior content will be there.